cupyx/scipy/signal/_bsplines.py

```diff
@@ -390,3 +390,40 @@ def qspline1d_eval(cj, newx, dx=1.0, x0=0):
         result += cj[indj] * _quadratic(newx - thisj)
     res[cond3] = result
     return res
+
+
+def spline_filter(Iin, lmbda=5.0):
+    """Smoothing spline (cubic) filtering of a rank-2 array.
+
+    Filter an input data set, `Iin`, using a (cubic) smoothing spline of
+    fall-off `lmbda`.
+
+    Parameters
+    ----------
+    Iin : array_like
+        input data set
+    lmbda : float, optional
+        spline smooghing fall-off value, default is `5.0`.
+
+    Returns
+    -------
+    res : ndarray
+        filterd input data
+
+    """
+    intype = Iin.dtype.char
+    hcol = cupy.asarray([1.0, 4.0, 1.0], 'f') / 6.0
+    if intype in ['F', 'D']:
+        Iin = Iin.astype('F')
```

Comment:
This is strange (both here and in scipy) : why does it only work in single precision complex. 

Also below, is there a major reason sepfir2d cannot natively support complex inputs?

Reply:
As you mention it, this is directly copied from SciPy, no modifications so far. We would need to research more about the sepfir implementation.

My guess is that all the implementations in _splines_mod.c.in (Or that template file in SciPy) only define implementations for floating point numbers and not for other data types